### ISSUE TITLE
ParsingException when sending temporary table data in CSV format

### ISSUE THREAD

**Issue by yiquanzhou** [ClickHouse/ClickHouse#23705]
**Describe the bug**
After upgrading from v20.3.19 to v21.3.4.25, the following HTTP request to Clickhouse does not work anymore:

`POST http://localhost:8123/?temp_table_format=CSV&temp_table_structure=id%20String%2Cname%20String`

The temp table structure is simply
```
id String,
name String
```

Headers:
`Content-Type: multipart/form-data; boundary=abc`

Request Body:
```
--abc
Content-Disposition: form-data; name="query"
Content-Length: 8

SELECT 1
--abc
Content-Disposition: form-data; name="temp_table"; filename="dummy"
Content-Length: 528889

"78817dce-b72e-4c18-8ea6-2f3c56be9d9f","service0"
"f9e2a876-1204-4ac5-a8fd-d415fbbb8a61","service1"
"aa4ce1e3-6eb9-4a59-95c4-a2b3f8c8678e","service2"
"0005b6ef-5966-4414-85b0-5d468e2a3d61","service3"
"9a0dfe2c-11b9-4a05-afca-ca49d092500f","service4"
"a4440b4b-895a-4523-9379-ba48c124e955","service5"
"34bcdf63-83c0-4921-a8da-2ea216fb8c63","service6"
... // 
"03c1e615-f6c6-4dee-a164-6c1d290c8e11","service9999"
--abc--
```
The complete file for the request body is [test-temp-table-10000.txt](https://github.com/ClickHouse/ClickHouse/files/6386887/test-temp-table-10000.txt)

The exception is 
```
Code: 27, e.displayText() = DB::ParsingException: Cannot parse input: expected ',' before: '\r\n"1571ebd5-3b79-4dd9-8be4-6e681e854807","service3879"\r\n"d4918bfa-2db4-4bd5-b4a3-16bbdd207b52","service3880"\r\n"57d6a2ae-215b-4e22-a26e-95c96b3de215","service388': 
Row 3881:
Column 0,   name: id,   type: String, parsed text: "<DOUBLE QUOTE>8534ce06-ce7b-470b-896b-b580c79ca38a<DOUBLE QUOTE>"
Column 1,   name: name, type: String, parsed text: "<DOUBLE QUOTE>service3878<DOUBLE QUOTE>"

Row 3882:
Column 0,   name: id,   type: String, parsed text: "e3878<DOUBLE QUOTE>"
ERROR: Line feed found where delimiter (,) is expected. It's like your file has less columns than expected.
And if your file have right number of columns, maybe it have unescaped quotes in values.

: While executing SourceFromInputStream: (at row 3882)
 (version 21.3.4.25 (official build))
```

It seems that the request body is not read entirely.  And it seems related to the size of the request body. If we only insert 1000 items into the temp table the request seems to work. [test-temp-table-1000.txt](https://github.com/ClickHouse/ClickHouse/files/6386902/test-temp-table-1000.txt)

`max_insert_block_size` is set to default value `1048545`.

**Comment by alexey-milovidov** [comment]
We have released the patches for version 21.3. You should install the version 21.3.8.

**Comment by yiquanzhou** [comment]
@alexey-milovidov  Thanks for the quick reply.
Tested with `21.3.8.76`, the issue is fixed.
